pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,145 @@ type ExternalNodeList struct {
+	// Represents the latest available observations of a SupportBundleCollection current state.
+	Conditions []SupportBundleCollectionCondition `json:"conditions"`
+}
+
+type SupportBundleCollectionConditionType string
+
+const (
+	// Started means Antrea Controller has created the internal resource for the SupportBundleCollectionCondition.
+	Started SupportBundleCollectionConditionType = "started"
+	// Collected means all Antrea Agents have successfully processed the SupportBundleCollectionCondition request.
+	Collected SupportBundleCollectionConditionType = "collected"
+)
+
+// SupportBundleCollectionCondition describes the state of a SupportBundleCollection at a certain point.
+type SupportBundleCollectionCondition struct {
+	// Type of StatefulSet condition.
+	Type SupportBundleCollectionConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status metav1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// A human-readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNodes struct {
+	Namespace string
+	// List the names of certain ExternalNodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain ExternalNodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+// BundleFileServer specifies the bundle file server information.
+type BundleFileServer struct {
+	// The URL of the bundle file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string `json:"url"`
+}
+
+// BundleServerAuthType defines the authentication type to access the BundleFileServer.
+type BundleServerAuthType string
+
+const (
+	APIKey      BundleServerAuthType = "apiKey"
+	BearerToken BundleServerAuthType = "bearerToken"
+)
+
+// BundleServerAuthConfiguration defines the authentication parameters that Antrea uses to access
+// the BundleFileServer.
+type BundleServerAuthConfiguration struct {
+	AuthType BundleServerAuthType `json:"authType"`
+	// AuthSecret is a Secret reference which stores the authentication value.
+	AuthSecret *v1.SecretReference `json:"authSecret"`
```

Comment:
I just found we didn't grant all secrets' read permission to antrea-controller, I feel it may be not a good practice to add all secret's read permission to antrea-controller for this feature. How do you think? @jianjuns @antoninbas

Reply:
I tend to agree. I am not sure what an alternative could be to achieve this though. If Antrea was running in its own Namespace, we could grant permission to read all Secrets in that Namespace, but Antrea typically runs inside kube-system.

Asking users to dynamically grant RBAC permission to the antrea-controller to read a specific Secret seems a bit cumbersome, but I can't think of anything else.

Reply:
I don't think we have a workaround for letting users to grant RBAC permissions to antrea-controller, because the token/apikey should not be exposed to everybody from the security perspective.

But for namespace choice, is it OK if we have a shared Namespace, which is only used to store supportbundle secret? User can create Secret in it, and antrea-controller get Secret in it? And we can define a special clusterrole in this namespace, user's service account can be bound to it?

Reply:
I assume we do not provide a file server in Antrea main repo? How about Nephe?

Probably we can assume typically the Secret is created in the file server or Nephe Namespace, and in the manual configuration case users may use a single Secret. And in all the cases, we require users or another solution to configure RBAC for controller to read the Secret?

Reply:
The file server is not provided in Antrea, and I agree that Nephe is a better choice to provide the file server and create role binding..